### PR TITLE
🚀Release To Preview

### LINKED ISSUE #44: 🚀Release To Preview
### Issues Released

- #12
- #10
- #12
- #5
- #18
- #16
- #23
- #9
- #28
- #27
- #30
- #35
- #38
- #41

### Perform Release ToDo List

- [x] Appropriate release label added to this issue.  Refer to the _**Release Labels**_ section below.
- [x] Release PR linked to this issue.
- [x] ~Manual QA Testing completed.~
- [x] All issues listed in the **_Issues Released_** section above.
- [x] All issues in the _**Issues Released**_ section are closed and all issue tasks are complete.
- [x] All unit tests have been executed locally and have passed. (Check out the appropriate release branch before running tests).
- [x] Version in project file updated (All changes made directly on the release branch).
- [x] Release notes created and added (All changes made directly on the release branch).
- [x] Pull request has been approved and merged into a release branch before performing the release. (Releases are performed on release branches, not preview release branches.)
- [x] Release to **_preview_** or **_production_** complete.  Refer to the **_Release Execution_** section below.

### Post-Release ToDo List

- [x] The release displayed in [GitHub Releases](https://github.com/KinsonDigital/BranchValidator/releases) is correct.
- [x] ~The [nuget package](https://www.nuget.org/packages/KinsonDigital.BranchValidator/) has been successfully deployed to the public repository.~
- [x] Announcement of release on [Twitter](https://twitter.com/KDCoder) verified. (Announcement should be performed automatically with the release)
- [x] Announcement has been pushed to the Discord channel.

### Additional Information:

**_<details closed><summary>Release Labels</summary>_**

| Release Types         | Label Used                                                                             |
|-----------------------|----------------------------------------------------------------------------------------|
| Preview Release       | https://github.com/KinsonDigital/BranchValidator/labels/%F0%9F%9A%80Preview%20Release         |
| Production Release    | https://github.com/KinsonDigital/BranchValidator/labels/%F0%9F%9A%80Production%20Release      |
| Hot Fix Release       | https://github.com/KinsonDigital/BranchValidator/labels/%F0%9F%9A%80Hot%20Fix%20Release       |
</details>

---

**_<details closed><summary>Unit Tests</summary>_**

Reasons for local unit test execution:
- Unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
- Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
</details>

---

**_<details closed><summary>Version Updating</summary>_**

The version can be updated by setting the values of the `<Version/>` and `<FileVersion/>` XML tags in the project file.

``` xml
<!--Syntax for preview releases-->
<Version>1.2.3-preview.4</Version>
<FileVersion>1.2.3-preview.4</FileVersion>

<!--Syntax for production releases-->
<Version>1.2.3</Version>
<FileVersion>1.2.3</FileVersion>
```
</details>

---

**_<details closed><summary>Release Notes</summary>_**

The release notes go into the appropriate folder:
- **Preview Release Notes** go into the **_~/Documentation/ReleaseNotes/PreviewReleases_** folder.
- **Production Release Notes** go into the **_~/Documentation/ReleaseNotes/ProductionReleases_** folder.

Release note file names must follow a particular syntax and are in markdown format so they can be added to the release.

#### Release Notes File Name Syntax:
- **Preview Release Notes:**
  - **Syntax:** Release-Notes-v\<_major_\>.\<_minor_\>.\<_patch_\>-preview.\<_prev-num_\>.md
  - **Example:** Release-Notes-v1.0.0-preview.4.md
- **Production Release Notes:**
  - **Syntax:** Release-Notes-v\<_major_\>.\<_minor_\>.\<_patch_\>.md
  - **Example:** Release-Notes-v1.0.0.md

Changes such as release notes and version updates should be committed to the same source branch in the pull request attached to this issue.
</details>

---

**_<details closed><summary>Release Execution</summary>_**

The release is performed by running one of the release workflows below:
- [Preview Release](https://github.com/KinsonDigital/BranchValidator/actions/workflows/preview-release.yml)
- [Production Release](https://github.com/KinsonDigital/BranchValidator/actions/workflows/qa-prod-releases.yml)
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Release Pull Request</h1>
<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

- [x] The **_base(target)_** branch for this pull request is a **_release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: release/v<major>.<minor>.<patch>
      Example: release/v1.2.3
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_release_** branch with the correct naming syntax in the following example.
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: release/v<major>.<minor>.<patch>
    Example: release/v1.2.3
    ```
  </details>

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the example below, using the correct version.
  <details closed><summary>Example</summary>
    
    ``` xml
    Syntax: 🚀Release To Preview - v<major>.<minor>.<patch>-preview.<prev-num>
    Example: 🚀Release To Preview - v1.2.3-preview.4
    ```
  </details>

- [x] A https://github.com/KinsonDigital/BranchValidator/labels/%F0%9F%9A%80Preview%20Release label has been added to this PR.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] ~A QA issue has been created and manual QA testing has been performed for the changes being released.~

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
No QA testing required and also not setup

**Review by KinsonDigitalAdmin — DISMISSED** [review]

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+<h2 style="font-weight:bold" align="center">Other 🪧</h2>
+<h5 align="center">(Includes anything that does not fit into the categories above)</h5>
+
+1. [#5](https://github.com/KinsonDigital/BranchValidator/issues/5) - Create logo for project.
+    - Create light and dark mode versions of the project logo.
+    - Added logo to the top of the project README file.
+2. [#18](https://github.com/KinsonDigital/BranchValidator/issues/18) - Updated and added issue and PR templates to the project.
+3. [#16](https://github.com/KinsonDigital/BranchValidator/issues/16) - Added code of conduct file to the project.
+4. [#23](https://github.com/KinsonDigital/BranchValidator/issues/23) - Fixed `release-todo-issue-template.yml` by removing character artifacts that were preventing the template from working properly.
+4. [#9](https://github.com/KinsonDigital/BranchValidator/issues/9) - Updated the name of the GitHub action from `BranchValidator` to `Branch Validator`
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md:61]: Re-number items.  There are two "4's"

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - This will make it a little more intuitive when searching for actions in the marketplace
+5. [#28](https://github.com/KinsonDigital/CASL/issues/28) - Updated the `release-todo-issue-template.yml` file.
+    - Added a new item to the **Perform Release ToDo List** section to check that the PR is approved and merged.
+    - Bolded a section of the last item in the **Perform Release ToDo List** section.
+6. [#27](https://github.com/KinsonDigital/CASL/issues/27) - Added branching documentation with branching documents.
+7. [#30](https://github.com/KinsonDigital/CASL/issues/30) - Updated project readme.
+    - Added a note explaining wny the action must be ran in Ubuntu with some links to some GitHub documentation resources.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md:68]: spelling/grammar
```suggestion
    - Added a note explaining why the action must run in Ubuntu with some links to some GitHub documentation resources.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+8. [#35](https://github.com/KinsonDigital/CASL/issues/35) - Updated the `preview-feature-pr-template.md` pull request template file.
+    - Rearranged the order of 2 tasks items.
+9. [#38](https://github.com/KinsonDigital/BranchValidator/issues/38) - Improve project README.
+10. [#41](https://github.com/KinsonDigital/BranchValidator/issues/41) - Made various improvements to the pull request templates by adding 2 new tasks as well as rearranging the order of of the tasks.  List of pull request template updated below:
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md:72]: Remove the extra "of" and added "s" to template.
```suggestion
10. [#41](https://github.com/KinsonDigital/BranchValidator/issues/41) - Made various improvements to the pull request templates by adding 2 new tasks as well as rearranging the order of the tasks.  List of pull request templates updated below:
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+1. [#12](https://github.com/KinsonDigital/BranchValidator/issues/12) - Added nuget package **KinsonDigital.KDActionUtils** version **v1.0.0-preview.1**.
+
+---
+
+<h2 style="font-weight:bold" align="center">Other 🪧</h2>
+<h5 align="center">(Includes anything that does not fit into the categories above)</h5>
+
+1. [#5](https://github.com/KinsonDigital/BranchValidator/issues/5) - Create logo for project.
+    - Create light and dark mode versions of the project logo.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md:56]: Grammar - updated to use past tense on both 'create' words.
```suggestion
1. [#5](https://github.com/KinsonDigital/BranchValidator/issues/5) - Created logo for project.
    - Created light and dark mode versions of the project logo.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    - This will make it a little more intuitive when searching for actions in the marketplace
+5. [#28](https://github.com/KinsonDigital/CASL/issues/28) - Updated the `release-todo-issue-template.yml` file.
+    - Added a new item to the **Perform Release ToDo List** section to check that the PR is approved and merged.
+    - Bolded a section of the last item in the **Perform Release ToDo List** section.
+6. [#27](https://github.com/KinsonDigital/CASL/issues/27) - Added branching documentation with branching documents.
+7. [#30](https://github.com/KinsonDigital/CASL/issues/30) - Updated project readme.
+    - Added a note explaining wny the action must be ran in Ubuntu with some links to some GitHub documentation resources.
+8. [#35](https://github.com/KinsonDigital/CASL/issues/35) - Updated the `preview-feature-pr-template.md` pull request template file.
+    - Rearranged the order of 2 tasks items.
+9. [#38](https://github.com/KinsonDigital/BranchValidator/issues/38) - Improve project README.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.2.md:71]: grammar - use past tense
```suggestion
9. [#38](https://github.com/KinsonDigital/BranchValidator/issues/38) - Improved project README.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — APPROVED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**